pkg/sdn/api/types.go

```diff
@@ -60,3 +60,33 @@ type NetNamespaceList struct {
 	unversioned.ListMeta
 	Items []NetNamespace
 }
+
+// EgressFirewallPolicy gives the type of an EgressFirewallRule
+type EgressFirewallPolicy string
+
+const (
+	EgressFirewallPolicyAllow EgressFirewallPolicy = "allow"
+	EgressFirewallPolicyDeny  EgressFirewallPolicy = "deny"
+)
+
+// EgressFirewallRule contains a single outgoing traffic firewall rule
+type EgressFirewallRule struct {
+	Policy EgressFirewallPolicy
+	From   string // Namespace name or empty
```

Comment:
> Do we need fallback rules? For quota we simply distribute the rules to all namespaces. For other places we create cluster scoped resources or use global config.

Can you point me to an example of where we have both namespace-scoped and cluster-scoped versions of a resource? As I understand it, this would require a second resource type (`ClusterEgressFirewall` or whatever) since a given resource type has to be either always namespaced or never namespaced, right?

(One possibility would be to add an array of `EgressFirewallRule` to the `ClusterNetwork` record?)

> If we think in the short term we don't need project admins editing egress rules, then let's just have no fallback, and require admins to specify them (we already effectively do this for quota, so again, no different).

Hm... I think if we were going to do this we'd have to at least add a global "deny egress by default" option, so that if the admin forgets to set up the firewall for a new project, then the project would default to being completely locked down rather than completely open.